File: src/main/java/com/adjorno/billib/rest/db/DayTrack.java
package com.adjorno.billib.rest.db;

import javax.persistence.*;
import java.sql.Date;

@Entity
@Table(name = "DAY_TRACK")
public class DayTrack {
    @Id
    @Column(name = "DAY")
    private Date mDay;

    @OneToOne
    @JoinColumn(name = "TRACK_ID")
    private Track mTrack;

    @Column(name = "DESCRIPTION")
    private String mDesc;

    public DayTrack() {
    }

    public DayTrack(Date day, Track track, String desc) {
        mDay = day;
        mTrack = track;
        mDesc = desc;
    }

    public Date getDay() {
        return mDay;
    }

    public void setDay(Date day) {
        mDay = day;
    }

    public Track getTrack() {
        return mTrack;
    }

    public void setTrack(Track track) {
        mTrack = track;
    }

    public String getDesc() {
        return mDesc;
    }

    public void setDesc(String desc) {
        mDesc = desc;
    }
}

File: src/main/java/com/adjorno/billib/rest/db/DayTrack.kt
package com.adjorno.billib.rest.db

import java.sql.Date
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.OneToOne
import jakarta.persistence.Table

@Entity
@Table(name = "DAY_TRACK")
data class DayTrack(
        @Id
        @Column(name = "_DAY")
        var day: Date? = null,

        @OneToOne
        @JoinColumn(name = "TRACK_ID")
        var track: Track? = null,

        @Column(name = "DESCRIPTION")
        var desc: String? = null
)
